neuraxle/pipeline.py

```diff
@@ -72,15 +72,16 @@ def transform(self, data_inputs: Any):
         """
         self.setup()
 
-        current_ids = self.hasher.hash(
-            current_ids=None,
-            hyperparameters=self.hyperparams,
-            data_inputs=data_inputs
-        )
-        data_container = DataContainer(current_ids=current_ids, data_inputs=data_inputs)
-        data_container = self._transform_core(data_container)
-
-        self.teardown()
+        try:
+            current_ids = self.hasher.hash(
+                current_ids=None,
+                hyperparameters=self.hyperparams,
+                data_inputs=data_inputs
+            )
+            data_container = DataContainer(current_ids=current_ids, data_inputs=data_inputs)
+            data_container = self._transform_core(data_container)
+        finally:
+            self.teardown()
```

Comment:
Since when do we teardown at the end of a pipeline? 

Please don't do this. We might want to continue the session. Teardown must be manual, or natural at the end of the program when the `BaseStep.__del__` method is called during garbage collection. 

Added issue #70. You can resolve this comment now already.

Reply:
not supposed to be in this pr